app/weeklyPicks/page.tsx

```diff
@@ -1,3 +1,172 @@
-export default function WeeklyPicks() {
-  return <h1>Weekly Picks</h1>;
+'use client';
+import React, { useState, useEffect } from 'react';
+import { Models } from 'appwrite';
+import { WeeklyPickButton } from '../../components/WeeklyPickButton/WeeklyPickButton';
+import { RadioGroup } from '../../components/RadioGroup/RadioGroup';
+import { Button } from '../../components/Button/Button';
+import {
+  getUserWeeklyPick,
+  createWeeklyPicks,
+  getNFLTeams,
+} from '@/api/apiFunctions';
+import { account } from '../../api/config';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+} from '../../components/Form/Form';
+
+/////// Edit to include the list of teams or from the API: ///////
+const teams = ['Vikings', 'Cowboys'] as const;
+//////////////////////////////////////////////////////////////////
+
+const FormSchema = z.object({
+  type: z.enum(teams, {
+    required_error: 'You need to select a team.',
+  }),
+});
+
+export default function WeeklyPickForm() {
+  const [NFLTeams, setNFLTeams] = useState<Models.Document[]>([]);
+  const [weeklyPicks, setWeeklyPicks] = useState<Models.Document>();
+
+  useEffect(() => {
+    async function fetchNFLTeams() {
+      const data = await getNFLTeams();
+      if (data) {
+        const response = data.documents;
+        setNFLTeams(response);
+      }
+    }
+
+    fetchNFLTeams();
+  }, []);
+
+  useEffect(() => {
+    async function fetchWeeklyPicks() {
+      const user = await account.get();
+
+      const userPicks = await getUserWeeklyPick({
+        userId: user.$id,
+        weekNumber: '6622c75658b8df4c4612',
+      });
+
+      const nflTeams = await getNFLTeams();
+
+      if (nflTeams && userPicks) {
+        const userResponse = userPicks;
+        const nflResponse = nflTeams.documents;
+
+        const findTeam = nflResponse.find(
+          (team) => team.$id === userResponse[user.$id].team,
+        );
+
+        setWeeklyPicks(findTeam);
+      }
+    }
+
+    fetchWeeklyPicks();
+  }, []);
+
+  const form = useForm<z.infer<typeof FormSchema>>({
+    resolver: zodResolver(FormSchema),
+  });
+
+  async function onSubmit(data: z.infer<typeof FormSchema>) {
+    try {
+      const teamSelect = data.type.toLowerCase();
+
+      let correct = false;
+
+      if (teamSelect === 'vikings') {
+        correct = true;
+      } else if (teamSelect === 'cowboys') {
+        correct = false;
+      }
+
+      if (typeof window !== 'undefined') {
```

Comment:
Why are we checking if window is not undefined? If the app mounts, window should always be defined.

Reply:
because this causes typescript to throw an error when changing this logic to == or === 'defined', and this conditional is necessary to keep localStorage from preventing the build to crash at the moment.